### PR TITLE
Parameter mixin

### PR DESCRIPTION
**Pull request by mkanoor**
Moved code from sources-api into the common gem for strong parameter check based on openapi.json

The original code was here
https://github.com/ManageIQ/sources-api/blob/0c096f39ee826b8269e79a28bc01475183ae2475/app/controllers/application_controller.rb#L74

### CODE REVIEW

**Comment by mkanoor** [comment]
@bdunne @Fryguy 
Please review

File: lib/manageiq/api/common/application_controller_mixins/parameters.rb [hunk 1]
```diff
@@ -0,0 +1,95 @@
+module ManageIQ
+  module API
+    module Common
+      module ApplicationControllerMixins
+        module Parameters
+          def params_for_create
+            # We already validate this with OpenAPI validator, that validates every request, so we shouldn't do it again here.
+            body_params.permit!
```
Comment (Fryguy) [inline lib/manageiq/api/common/application_controller_mixins/parameters.rb]: Now that you have openapi_enabled = false...it's possible we could now be permitting bad params to your non-openapi_enabled controllers... cc @bdunne

Comment (Fryguy) [reply lib/manageiq/api/common/application_controller_mixins/parameters.rb]: This probably goes for almost all of the methods in here.

File: spec/dummy/app/controllers/api/v1x0.rb [hunk 2]
```diff
@@ -21,5 +21,22 @@ def show
         render :json => {:id => request_path_parts["primary_collection_id"]}.to_json
       end
     end
+
+    class PersonsController < ApplicationController
+      def create
+        params_for_create
```
Comment (Fryguy) [inline spec/dummy/app/controllers/api/v1x0.rb]: What does this test besides just running the methods?

File: lib/manageiq/api/common/application_controller_mixins/parameters.rb [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+            # :limit & :offset can be passed in for pagination purposes, but shouldn't show up as params for filtering purposes
+            @safe_params_for_list ||= params.merge(params_for_polymorphic_subcollection).permit(*permitted_params, :filter => {})
+          end
+
+          def permitted_params
+            api_doc_definition.all_attributes + [:limit, :offset] + [subcollection_foreign_key]
+          end
+
+          def subcollection_foreign_key
+            "#{request_path_parts["primary_collection_name"].singularize}_id"
```
Comment (Fryguy) [inline lib/manageiq/api/common/application_controller_mixins/parameters.rb]: I know it's not from this PR, but we a more Rails oriented way to do this...there's should be a helper that gets you this string

File: lib/manageiq/api/common/application_controller_mixins/parameters.rb (hunk 1 above)
Comment (mkanoor) [reply lib/manageiq/api/common/application_controller_mixins/parameters.rb]: @Fryguy But you would be calling these methods only from openapi enabled controllers. A non openapi enabled controller doesn't have to call these methods

File: lib/manageiq/api/common/application_controller_mixins/parameters.rb [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+          end
+
+          def params_for_polymorphic_subcollection
+            return {} unless subcollection?
+            return {} unless reflection = primary_collection_model&.reflect_on_association(request_path_parts["subcollection_name"])
+            return {} unless as = reflection.options[:as]
+            {"#{as}_type" => primary_collection_model.name, "#{as}_id" => request_path_parts["primary_collection_id"]}
```
Comment (Fryguy) [inline lib/manageiq/api/common/application_controller_mixins/parameters.rb]: same here - I know it's not from this PR, but we a more Rails oriented way to do this...there's should be a helper that gets you this string  cc @bdunne

File: spec/dummy/app/controllers/api/v1x0.rb (hunk 2 above)
Comment (mkanoor) [reply spec/dummy/app/controllers/api/v1x0.rb]: @Fryguy It just calls the strong parameters check and if its bad we get a 400 back.

File: lib/manageiq/api/common/application_controller_mixins/parameters.rb (hunk 1 above)
Comment (Fryguy) [reply lib/manageiq/api/common/application_controller_mixins/parameters.rb]: No, but it's not uncommon from a controller to call methods like `params_for_create`, which would now be available to them.  At a minimum maybe these methods should raise an Exception unless openapi_enabled?

File: spec/dummy/app/controllers/api/v1x0.rb (hunk 2 above)
Comment (Fryguy) [reply spec/dummy/app/controllers/api/v1x0.rb]: Still feels like there should be a render call or something...Standalone it doesn't feel like a realistic looking tests.

Comment (mkanoor) [reply spec/dummy/app/controllers/api/v1x0.rb]: @Fryguy Please review I have added a check for openapi.

**Pull request merged**